pyaerocom/io/pyaro/read_pyaro.py

```diff
@@ -0,0 +1,292 @@
+from __future__ import annotations
+
+import logging
+from copy import deepcopy
+from typing import NewType
+
+import numpy as np
+from pyaro import list_timeseries_engines, open_timeseries
+from pyaro.timeseries import Data, Reader, Station
+from pyaro.timeseries.Wrappers import VariableNameChangingReader
+
+from pyaerocom.io.pyaro.pyaro_config import PyaroConfig
+from pyaerocom.io.readungriddedbase import ReadUngriddedBase
+from pyaerocom.tstype import TsType
+from pyaerocom.ungriddeddata import UngriddedData
+
+# TODO: Add possibility to filter after reading (if possible)
+"""
+TODO: Features to add:
+- instrument name
+- station_id?
+- filename
+- revision date
+- data level
+- PI
+
+
+- Make code faster...
+"""
+
+"""
+TODO: Features to check
+- ts_type (freq)
+"""
+
+logger = logging.getLogger(__name__)
+
+
+Metadata = NewType("Metadata", dict[str, dict[str, str | list[str]]])
```

Comment:
This is a cool idea and probably should be used in other parts of the code.

Reply:
Thanks :) And yes, it makes the code much easier to read